### PR TITLE
Set storage type combox for different cluster type

### PR DESCRIPTION
**Pull request by lcadzy**

### CODE REVIEW

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/ui/SparkSubmissionJobUploadStorageCtrl.kt [hunk 1]
```diff
@@ -148,22 +150,79 @@ abstract class SparkSubmissionJobUploadStorageCtrl(val view: SparkSubmissionJobU
...
+                val adlsTitle = view.storagePanel.adlsCard.title
+
+                val clusterDetail = getClusterDetail()
+                view.storagePanel.storageTypeComboBox.selectedItem = null
+                view.storagePanel.storageTypeComboBox.model = when (clusterDetail) {
+                    is ClusterDetail ->{
+                        //get storageaccount may get HDIExpcetion for null value
+                        var storageAccount = try {
+                            clusterDetail.storageAccount
+                        }catch(ex: HDIException)
```
Comment (wezhang) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/ui/SparkSubmissionJobUploadStorageCtrl.kt]:
```suggestion
                        } catch (ignored: HDIException)
```

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/ui/SparkSubmissionJobUploadStorageCtrl.kt [hunk 2]
```diff
@@ -148,22 +150,79 @@ abstract class SparkSubmissionJobUploadStorageCtrl(val view: SparkSubmissionJobU
...
-                view.storagePanel.storageTypeComboBox.selectedItem = view.storagePanel.sparkInteractiveSessionCard.title
-            } else {
-                view.storagePanel.storageTypeComboBox.selectedItem = view.storagePanel.clusterDefaultStorageCard.title
+    fun setDefaultStorageType(checkEvent:StorageCheckEvent) {
+        synchronized(view.storagePanel) {
+            if (checkEvent is StorageCheckSelectedClusterEvent) {
+
+                //check cluster type then reset storage combox
+                val defaultStorageTitle = view.storagePanel.clusterDefaultStorageCard.title
+                val sessionTitle = view.storagePanel.sparkInteractiveSessionCard.title
```
Comment (wezhang) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/ui/SparkSubmissionJobUploadStorageCtrl.kt]: `helpSessionTitle` may help remember it.

```suggestion
                val helpSessionTitle = view.storagePanel.sparkInteractiveSessionCard.title
```

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/ui/SparkSubmissionJobUploadStorageCtrl.kt [hunk 3]
```diff
@@ -148,22 +150,79 @@ abstract class SparkSubmissionJobUploadStorageCtrl(val view: SparkSubmissionJobU
...
+
+                    is HDInsightLivyLinkClusterDetail, is HDInsightAdditionalClusterDetail -> ImmutableComboBoxModel(arrayOf(
+                            azureBlobTitle,
+                            adlsTitle,
+                            sessionTitle)).apply {
+                        selectedItem = sessionTitle
+                    }
+
+                    is SqlBigDataLivyLinkClusterDetail -> {
+                        ImmutableComboBoxModel<String>(arrayOf(
```
Comment (wezhang) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/ui/SparkSubmissionJobUploadStorageCtrl.kt]:
```suggestion
                        ImmutableComboBoxModel(arrayOf(
```

**Comment by lcadzy** [comment]
#2484

**Comment by lcadzy** [comment]
#2316

**Pull request merged**